astropy/table/serialize.py

```diff
@@ -336,7 +336,14 @@ def _construct_mixin_from_obj_attrs_and_info(obj_attrs, info):
         mixin.info.name = info["name"]
         return mixin
 
-    if cls_full_name not in __construct_mixin_classes:
+    # We translate locally created skyoffset frames and treat all
+    # built-in frames as known.
+    if cls_full_name.startswith("abc.SkyOffset"):
+        cls_full_name = "astropy.coordinates.SkyOffsetFrame"
+    elif (
+        cls_full_name not in __construct_mixin_classes
+        and not cls_full_name.startswith("astropy.coordinates.builtin_frames")
```

Comment:
I did not really want to bother to add all the built-in frames explicitly to the white list. We may want to re-think the list more generally. E.g., anything starting with `astropy.coordinates` probably best is gotten from that directly, since that `__init__` is much less likely to contain any other class that might be used to wreck mayhem.